src/app/screens/BookDetail/components/CommentList/index.js

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react';
+import { View, Text, FlatList, TouchableOpacity } from 'react-native';
+
+import Comment from './components/Comment';
+import styles from './styles';
+
+class CommentList extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = { isViewedAll: false };
+    }
+    renderItem = ({item}) =>  <Comment username={item.user.name}  userImageSource={item.user.imageSource} comment={item.comment}/>;
+
+    keyExtractor = (item) => `${item.id}`;
+
+    onPressHandler = () => (this.setState(previousState => ({ isViewedAll: true })));
+    render() {
+        const {comments} = this.props;
+        return (
+            <FlatList
+            data={ (this.state.isViewedAll) ? comments.filter((x,i) => [0,1,2,3,4].includes(i)) : comments.filter((x,i) => [0,1].includes(i))}
```

Comment:
spaces

Reply:
Destruct the `this.state` object as you do with `this.props`:
```
const { isViewedAll } = this.state;
data={ isViewedAll ? comments.filter((x,i) => [0,1,2,3,4].includes(i)) : comments.filter((x,i) => [0,1].includes(i))}
```
Also, your are repeating this `comments.filter((x,i) => ARRAY.includes(i))`. I think you can extract this functions and pass via params the `comments` array and the `indexs` array ([0,1,2,3,4] and [0,1])